jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py

```diff
@@ -2,50 +2,172 @@
 __license__ = "Apache-2.0"
 
 import copy
-from typing import Dict
+import datetime
+import functools
+from typing import Dict, Optional
 
-from jina import requests, DocumentArray, Executor
-from jina.logging.logger import JinaLogger
+from jina import DocumentArray, Executor, requests
+from jina_commons import get_logger
 
 try:
     from jinahub.indexers.searcher.FaissSearcher import FaissSearcher
-except:
-    from jina_executors.indexers.searcher.FaissSearcher.faiss_searcher import FaissSearcher
+except ImportError:  # noqa: E722
+    from jina_executors.indexers.searcher.FaissSearcher.faiss_searcher import (
+        FaissSearcher,
+    )
 
 try:
-    from jinahub.indexers.storage.PostgreSQLStorage import (
-        PostgreSQLStorage,
-    )
-except:
-    from jina_executors.indexers.storage.PostgreSQLStorage import (
-        PostgreSQLStorage,
-    )
+    from jinahub.indexers.storage.PostgreSQLStorage import PostgreSQLStorage
+except ImportError:  # noqa: E722
+    from jina_executors.indexers.storage.PostgreSQLStorage import PostgreSQLStorage
 
 
 class FaissPostgresSearcher(Executor):
-    """A Compound Indexer made up of a FaissSearcher (for vectors) and a Postgres Indexer"""
+    """A Compound Indexer made up of a FaissSearcher (for vectors) and a Postgres
+    Indexer
+
+    :param dump_path: a path to a dump folder containing
+    the dump data obtained by calling jina_commons.dump_docs
+    :param startup_sync: whether to use the dump
+     function of PostgreSQLStorage, when dump_path is not provided
+    :param startup_sync_args: the arguments to be passed to the self.sync call on
+    startup (if startup_sync)
+    """
 
     def __init__(
         self,
-        dump_path=None,
+        dump_path: Optional[str] = None,
+        startup_sync: bool = False,
+        startup_sync_args: Optional[None] = None,
         **kwargs,
     ):
         super().__init__(**kwargs)
-        # when constructed from rolling update the dump_path is passed via a runtime_arg
+        self.logger = get_logger(self)
+
+        self.total_shards = self.runtime_args.parallel
```

Comment:
When using Kubernetes, parallel and replicas will always be set to 1 for each deployed execuror. For now I would suggest to create a new parameter `total_shards` instead.

Reply:
Ah yes, good catch. I added some code to handle that.